readthedocs/api/v2/utils.py

```diff
@@ -166,18 +170,23 @@ def delete_versions_from_db(project, version_data):
         identifier__in=versions_branches,
     )
 
-    ret_val = set(to_delete_qs.values_list('slug', flat=True))
-    if ret_val:
+    deleted_active_versions = set(
+        to_delete_qs.filter(active=True).values_list('slug', flat=True)
+    )
```

Comment:
This is confusing to me. These are not "deleted active versions". We are not deleting them in this function. These are "versions that we have marked as `active=True` in our DB but they don't exist anymore in the remote repository"

Reply:
Also, I think I'll feel more comfortable if we create a helper (eg. `get_remote_deleted_versions(active=False)`) for this instead of adding this in the middle of another function.

Reply:
The more I think about this, the less I like it. This function is called `delete_versions_from_db` and return 2 lists. One list are versions that are _in fact_ deleted from the DB, but the other list are versions that _are not deleted from DB_. Super confusing to me.